elastic/kibana
# [Fleet] Add packages to global search results

## Summary

Add packages to the global search's results and link through to the integrations details page. Closes #86171.

## How to test

1. Start Kibana + ES on basic
2. Type "Li" in the search bar
3. You should see the "Linux" package included in the search results
4. Click on the search result
5. You should now see the details of the Linux package as provided by the Integrations plugin

## To reviewers

* ~The current implementation creates a new request to fetch all the packages from EPR _every time_. This can be improved in a few different ways: we can only make a new request every 30 seconds (i.e., cache the first result), we can work with EPR to add an improved search API such that user input can be used to search for packages directly~
* I am not certain what else we would want to add here, which pages in particular would be useful to have as search results? Currently both fleet and integration top-level apps appear in the search results.
* Please see this issue regarding URLs in fleet that should be considered: https://github.com/elastic/kibana/issues/96134

## Screenshots

<details>
<summary>Gif</summary>

![linux-pkg-fleet](https://user-images.githubusercontent.com/8155004/122075388-54040600-cdfa-11eb-9bcb-22083fbfcb20.gif)

</details>

## Release note

We added fleet packages to the list of items that can be searched for via the global search bar in Kibana.


### Checklist
